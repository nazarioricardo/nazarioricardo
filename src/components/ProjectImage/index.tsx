import Image from "next/image";
import urlBuilder from "@sanity/image-url";
import client from "@nr/sanity/client";
import styles from "./styles.module.css";
import { SyntheticEvent, useState } from "react";
// https://www.sanity.io/answers/adding-images-to-post-content-using-portable-text-and-sanity-image-url-builder-

type ProjectImageProps = {
  value: {
    alt: string;
    asset: {
      _ref: string;
      _type: string;
    };
  };
};

function ProjectImage({ value }: ProjectImageProps) {
  const [didLoadImage, setDidLoadImage] = useState(false);
  return (
    <div className={styles.projectImage}>
      <Image
        className={`${styles.image} ${didLoadImage && styles.loadedImage}`}
        src={urlBuilder(client)
          .image(value)
          .width(2000)
          .fit("fill")
          .auto("format")
          .url()}
        alt={value.alt || "Post Image"}
        loading="lazy"
        width={500}
        height={500}
        quality={100}
        onLoad={(event: SyntheticEvent<HTMLImageElement, Event>) => {
          const { currentTarget } = event;

          // next/image use an 1x1 px git as placeholder. We only want the onLoad event on the actual image
          if (currentTarget.src.indexOf("data:image/gif;base64") < 0) {
            setDidLoadImage(true);
          }
        }}
      />
    </div>
  );
}

export default ProjectImage;
