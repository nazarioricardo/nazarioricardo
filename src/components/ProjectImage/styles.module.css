.projectImage {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.image {
  border-radius: 100px;
}

@media screen and (max-width: 600px) {
  .image {
    width: 100%;
    height: auto;
  }
}
