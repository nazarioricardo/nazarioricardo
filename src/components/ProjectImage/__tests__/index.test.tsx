import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProjectImage from "../index";

const mockProps = {
  value: {
    alt: "Test Image",
    asset: {
      _ref: "image-Tb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000-jpg",
      _type: "image",
    },
  },
};

jest.mock("@sanity/client", () => ({
  createClient: jest.fn(),
}));

describe("ProjectImage", () => {
  test("renders ProjectImage component", () => {
    render(<ProjectImage {...mockProps} />);
    const imgElement = screen.getByAltText(mockProps.value.alt);
    expect(imgElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ProjectImage {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
