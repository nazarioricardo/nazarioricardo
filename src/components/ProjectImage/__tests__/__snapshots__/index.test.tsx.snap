// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectImage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="projectImage"
  >
    <img
      alt="Test Image"
      class="image false"
      data-nimg="1"
      decoding="async"
      height="500"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fcdn.sanity.io%2Fimages%2Fundefined%2Fundefined%2FTb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000.jpg%3Fw%3D2000%26fit%3Dfill%26auto%3Dformat&w=1080&q=100"
      srcset="/_next/image?url=https%3A%2F%2Fcdn.sanity.io%2Fimages%2Fundefined%2Fundefined%2FTb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000.jpg%3Fw%3D2000%26fit%3Dfill%26auto%3Dformat&w=640&q=100 1x, /_next/image?url=https%3A%2F%2Fcdn.sanity.io%2Fimages%2Fundefined%2Fundefined%2FTb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000.jpg%3Fw%3D2000%26fit%3Dfill%26auto%3Dformat&w=1080&q=100 2x"
      style="color: transparent;"
      width="500"
    />
  </div>
</DocumentFragment>
`;
